File: get_weather.py
import dbinfo
import APIinfo
import sqlalchemy as sqla
from sqlalchemy import Table, Column, Integer, Float, String, MetaData, DateTime, create_engine
from sqlalchemy.orm import sessionmaker
import requests
from datetime import datetime
import json
import time
import traceback

url = "https://api.openweathermap.org/data/2.5/onecall?lat=%s&lon=%s&appid=%s&units=metric" % (APIinfo.DUBLIN_LATITUDE, APIinfo.DUBLIN_LONGITUDE, APIinfo.WEATHER_APIKEY)
engine = create_engine("mysql+mysqlconnector://{}:{}@{}:{}/{}".format(dbinfo.USER, dbinfo.PASSWORD, dbinfo.URI, dbinfo.PORT, dbinfo.DB), echo=True)

def main():
    while True:
        try:
            response = requests.get(url)
        
            store_weather(json.loads(response.text))
        
            time.sleep(30*60)
        except:
            print(traceback.format_exc())

#Parses weather obtained from API request
def parseWeather(obj):
    return {'time' : datetime.fromtimestamp(int(obj['dt'])),
            'temp' : obj['temp'],
            'humidity' : obj['humidity'],
            'wind_speed' : obj['wind_speed'],
            'main' : obj['weather'][0]['main'],
            'description' : obj['weather'][0]['description'],
            'visibility' : obj['visibility']
        }

#stores data in weather database
def store_weather(json_data):
    metadata = sqla.MetaData(bind=engine)
    weather = sqla.Table('weather', metadata, autoload=True)

    engine.execute(weather.insert(), parseWeather(json_data['current']))

main()

File: static/get_weather.js
function get_current_weather(id) {
    fetch("/weather").then(response => response.json())
        .then(function(data) {
        console.log(data[0]);
        var content = "";
        var date = new Date(data[0]['time']);
        content = "<h2>" + data[0]['main'] + "</h2>" +
            "<ul><li>" + data[0]['description'] + 
            "</li><li>Temperature: " + data[0]['temp'] + 
            "C</li><li>Humidity: " + data[0]['humidity'] +
            "%</li><li>Visibility: " + data[0]['visibility'] + 
            "m</li><li>Wind Speed: " + data[0]['wind_speed'] +
            "m/s</li></ul><h4>Time Taken: " + date + "</h4>"
        document.getElementById(id).innerHTML = content;
    })
    .catch(err => {
        console.log("Unable to collect/format current weather data");
    })
}
console.log('huh');